### PR TITLE
[dataset] remove mType from Dataset

### PR DESCRIPTION
**Pull request by bukepo**
This commit removes the mType field from the Dataset class to save RAM.

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 👍

**Comment by jwhui** [comment]
> This commit removes the mType field from the Dataset class to save RAM.

Size report indicates that there are no RAM savings - due to padding, I believe. However, there is marginal savings in code size.

File: src/core/meshcop/dataset.hpp [hunk 1]
```diff
@@ -615,10 +615,8 @@ class Dataset
     /**
      * This constructor initializes the object.
      *
-     * @param[in]  aType       The type of the dataset, active or pending.
-     *
      */
-    explicit Dataset(Type aType);
+    explicit Dataset(void);
```
Comment (jwhui) [inline src/core/meshcop/dataset.hpp:619]: Did you intend to remove the following? I don't see it removed in this PR. https://github.com/openthread/openthread/blob/71e00eb20f14992ecbbecab8e0c79cf5aa76bb9d/src/core/meshcop/dataset.hpp#L926

Comment (abtink) [inline src/core/meshcop/dataset.hpp:619]: Don't need the `explicit` anymore?

**Comment by bukepo** [comment]
> > This commit removes the mType field from the Dataset class to save RAM.
> 
> Size report indicates that there are no RAM savings - due to padding, I believe. However, there is marginal savings in code size.

Yeah, I guess so..

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**